app/src/main/java/com/scouts/kitchenplaner/model/usecases/EditMealPlan.kt

```diff
@@ -18,23 +18,39 @@ package com.scouts.kitchenplaner.model.usecases
 
 import com.scouts.kitchenplaner.datalayer.repositories.ProjectRepository
 import com.scouts.kitchenplaner.datalayer.repositories.RecipeManagementRepository
+import com.scouts.kitchenplaner.datalayer.repositories.RecipeRepository
 import com.scouts.kitchenplaner.model.entities.MealSlot
 import com.scouts.kitchenplaner.model.entities.Project
-import com.scouts.kitchenplaner.model.entities.Recipe
+import com.scouts.kitchenplaner.model.entities.RecipeStub
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
 import javax.inject.Inject
 
 class EditMealPlan @Inject constructor(
     private val recipeManagementRepository: RecipeManagementRepository,
+    private val recipeRepository: RecipeRepository,
     private val projectRepository: ProjectRepository
 ) {
-    suspend fun selectRecipeForMealSlot(project: Project, mealSlot: MealSlot, recipe: Recipe) {
+    suspend fun selectMainRecipeForMealSlot(project: Project, mealSlot: MealSlot, recipe: RecipeStub) {
         recipeManagementRepository.chooseMainRecipeForMealSlot(
             project.id,
-            recipe.id,
+            recipe.id ?: return,
```

Comment:
RecipeStub::id is Nullable for some reason. However the repository method requires a Non-Null Long as a parameter (as it doesn't really make sense to explicitely store that no recipe has been assigned). If the passed RecipeStub has a Null id, the repository thus can't be called with any kind of useful parameter and we instead return out of the model method

Reply:
If the question was just "What does ```xyz ?: return``` do?", then the short answer is that if xyz is not null, it is passed as the parameter, otherwise we return out of the method (in this case ```selectMainRecipeForMealSlot()```)